oak_idl/src/lib.rs

```diff
@@ -31,6 +31,20 @@ pub enum Error {
     InvalidResponse,
     /// The method id provided for the invocation was not implemented by the server.
     InvalidMethodId,
+    /// An error occured while invoking the method on the server.
+    MethodError(MethodError),
```

Comment:
I think the extra layer of error does not really add much (which is why in 457e70e6aa67ce674d95a681c6b2a0f28f603436 I just removed the one I originally added). In the end the caller will probably just log something and move on, so having a complex error hierarchy does not really help much IMO. I would reserve this pattern for when some class of error comes from a different library, and we don't want to just convert to our own error space, but we don't have that problem here (yet).

And I think we should probably just switch to the gRPC error codes at some point, and also allow the method to return any of them (e.g. it may return NOT_FOUND directly if it needs to, which will be useful).

Reply:
Alright, can do. I personally disagree and think that a lot of meaning is lost in gRPC's approach, but can go that way. In that case we should probably indeed switch to gRPC codes. It's worth pointing out though that even in gRPC's case, some codes are never meant to originate in the client. Hence we'd either need to maintain hierarchy like this, or rely on the method implementation to never send those errors.